Core/src/org/sleuthkit/autopsy/modules/iOS/TextMessageAnalyzer.java

```diff
@@ -62,24 +72,33 @@ void findTexts(IngestJobContext context) {
             if (absFiles.isEmpty()) {
                 return;
             }
-            for (AbstractFile AF : absFiles) {
+            for (AbstractFile file : absFiles) {
                 try {
-                    jFile = new java.io.File(Case.getCurrentCase().getTempDirectory(), AF.getName().replaceAll("[<>%|\"/:*\\\\]", ""));
-                    ContentUtils.writeToFile(AF, jFile, context::dataSourceIngestIsCancelled);
+                    jFile = new java.io.File(Case.getCurrentCase().getTempDirectory(), file.getName().replaceAll("[<>%|\"/:*\\\\]", ""));
                     dbPath = jFile.toString(); //path of file as string
-                    fileId = AF.getId();
+                    ContentUtils.writeToFile(file, jFile, context::dataSourceIngestIsCancelled);
+                    fileId = file.getId();
```

Comment:
The call to AbstractFile.getId needs to occur before the call to ContentUtils.writeToFile.